pentaho-osgi-utils/impl/src/main/java/org/pentaho/osgi/impl/KarafFeatureWatcherImpl.java

```diff
@@ -263,11 +267,13 @@ private String getFeatureReport( FeaturesService featuresService, BundleService
     }
 
     // For this feature, we list its non active bundles with additional information
-    if ( feature.getBundles() != null ) {
+    if ( null == bundleService ) {
+      featureReport += "Cannot list bundles; bundleService not available";
+    } else if ( feature.getBundles() != null ) {
       boolean first = true;
       for ( BundleInfo bundleInfo : feature.getBundles() ) {
         Bundle bundle = bundleContext.getBundle( bundleInfo.getLocation() );
-        if ( bundleService.getInfo( bundle ).getState() != BundleState.Active ) {
+        if ( null != bundle && bundleService.getInfo( bundle ).getState() != BundleState.Active ) {
```

Comment:
Do we want a separate log line to capture that the bundle is null?

basically a print of of `bundleInfo.getLocation()`

Reply:
Sure, makes sense.